crates/ruff_linter/src/rules/flake8_import_conventions/rules/unconventional_import_alias.rs

```diff
@@ -69,6 +70,11 @@ pub(crate) fn unconventional_import_alias(
         return;
     };
 
+    let normalized_alias = expected_alias.nfkc().collect::<String>();
```

Comment:
It would be better to do this normalization when loading the settings. We otherwise pay the cost for every import statement

Reply:
I'm not sure if I've worked with settings before, how would it be implemented? Via these?

- `crates\ruff_workspace\src\options.rs`
- `crates\ruff_linter\src\rules\flake8_import_conventions\settings.rs`

Reply:
Ideally, we'd validate the settings when loading the configuration. The idea is to rename the [`into_settings`](https://github.com/astral-sh/ruff/blob/8c0743df97dc7afa1deab7506b322551f2e392bb/crates/ruff_workspace/src/options.rs#L1653-L1674) method to `try_into_settings` and change its return type to `Result`, similar to 

https://github.com/astral-sh/ruff/blob/8c0743df97dc7afa1deab7506b322551f2e392bb/crates/ruff_workspace/src/options.rs#L1416-L1428